python/lsst/sconsUtils/vcs/git.py

```diff
@@ -23,11 +23,13 @@ def guessVersionName():
     if not os.path.exists(".git"):
         state.log.warn("Cannot guess version without .git directory; version will be set to 'unknown'.")
         return "unknown"
+    desc = utils.runExternal("git describe --tags --always", fatal=True)
+    desc = desc.strip()
+
     status = utils.runExternal("git status --porcelain --untracked-files=no", fatal=True)
     if status.strip():
-        raise RuntimeError("Error with git version: uncommitted changes")
-    desc = utils.runExternal("git describe --tags --always", fatal=True)
-    return desc.strip()
+        desc += " *"
```

Comment:
Not sure if this is important but [python versions](https://packaging.python.org/en/latest/specifications/version-specifiers/#) shouldn't have spaces or end in non-digit or non-number. We likely don't care because these version strings will fail validation anyhow. In some sense `*` is a bit ambivalent as to what it means and `-dirty` is more explicit (and I don't think it matters if that turns up in fingerprint).

@ktlim do you have an opinion on this?

Reply:
ok changing the code for version but not for fingerprint.

Reply:
I can't think of a reason why we wouldn't use `-dirty` for fingerprint as well. Surely people aren't parsing that for the `*`?